src/SprykerEco/Zed/Easycredit/Business/Parser/ResponseParser.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Easycredit\Business\Parser;
+
+use Generated\Shared\Transfer\EasycreditApprovalTextResponseTransfer;
+use Generated\Shared\Transfer\EasycreditInitializePaymentResponseTransfer;
+use Generated\Shared\Transfer\EasycreditInterestAndAdjustTotalSumResponseTransfer;
+use Generated\Shared\Transfer\EasycreditOrderConfirmationResponseTransfer;
+use Generated\Shared\Transfer\EasycreditPreContractualInformationAndRedemptionPlanResponseTransfer;
+use Generated\Shared\Transfer\EasycreditQueryCreditAssessmentResponseTransfer;
+use Generated\Shared\Transfer\EasycreditResponseTransfer;
+
+class ResponseParser implements ResponseParserInterface
+{
+    protected const KEY_PAYMENT_IDENTIFIER = 'tbVorgangskennung';
+    protected const KEY_ALLGEMEINE_VORGANGSDATEN = 'allgemeineVorgangsdaten';
+    protected const KEY_URL_VORVERTRAGLICHE_INFORMATIONEN = 'urlVorvertraglicheInformationen';
+    protected const KEY_TILGUNGSPLAN_TEXT = 'tilgungsplanText';
+    protected const KEY_WS_MESSAGES = 'wsMessages';
+    protected const KEY_MESSAGES = 'messages';
+    protected const KEY_KEY = 'key';
+    protected const VALUE_SUCCESS_CONFIRMATION = 'BestellungBestaetigenServiceActivity.Infos.ERFOLGREICH';
+    protected const KEY_RATENPLAN = 'ratenplan';
+    protected const KEY_ZINSEN = 'zinsen';
+    protected const KEY_ANFALLENDE_ZINSEN = 'anfallendeZinsen';
+    protected const KEY_ENTSCHEIDUNG = 'entscheidung';
+    protected const KEY_ENTSCHEIDUNG_SERGEBNIS = 'entscheidungsergebnis';
+    protected const KEY_TEXT_IDENTIFIER = 'zustimmungDatenuebertragungPaymentPage';
+
+    /**
+     * @param \Generated\Shared\Transfer\EasycreditResponseTransfer $easycreditResponseTransfer
+     *
+     * @return \Generated\Shared\Transfer\EasycreditInitializePaymentResponseTransfer
+     */
+    public function parseInitializePaymentResponse(EasycreditResponseTransfer $easycreditResponseTransfer): EasycreditInitializePaymentResponseTransfer
+    {
+        $payload = $easycreditResponseTransfer->getBody();
+
+        $easycreditInitializePaymentResponseTransfer = new EasycreditInitializePaymentResponseTransfer();
+        $easycreditInitializePaymentResponseTransfer->setSuccess(false);
+
+        if ($easycreditResponseTransfer->getError() === null) {
```

Comment:
If we have no errors in transfer we return fail response transfer. I didn't get an idea.